### ISSUE TITLE
No 'Access-Control-Allow-Origin' Header is present

### ISSUE THREAD

**Issue by BaileyJM02** [CachetHQ/Cachet#2947]
I have the following code:
```JS
var data = JSON.stringify({
  "value": 2 
});

var xhr = new XMLHttpRequest();
xhr.withCredentials = false; // Also tired with true

xhr.addEventListener("readystatechange", function () {
  if (this.readyState === this.DONE) {
    console.log(this.responseText);
  }
});



xhr.open("POST",
  "http://status.bailey.guru/api/v1/metrics/1/points"
);

xhr.setRequestHeader("Content-Type", "application/json");
xhr.setRequestHeader("X-Cachet-Token", "token....");

xhr.send(data);
```

I'm trying to add a point to one of my metrics however I am returned with the following error:
```
Failed to load http://status.bailey.guru/api/v1/metrics/1/points: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'http://vps.bailey.guru' is therefore not allowed access.
```
I think this is a bug related to #2473 as that is the error I got before adding `xhr.setRequestHeader("Content-Type", "application/json");`.

I've tried for the last 3-4 hours to fix it but to no eval. Can anyone help?

**Comment by uxen-ab** [comment]
Have you tried to add a header 'Origin' in your request?
Otherwise come on Slack we'll be glad to help you!

**Comment by jbrooksuk** [comment]
https://docs.cachethq.io/v1.0/docs/cors

**Comment by BaileyJM02** [comment]
Yeah, I tried that but it didn't work. Instead I've just used PHP. Though a fix would be nice. 😀

**Comment by jbrooksuk** [comment]
🤔

**Comment by jbrooksuk** [comment]
I think you need to add authentication to the request.

**Comment by BaileyJM02** [comment]
How would I do that then? I thought the token was enough.

**Comment by jbrooksuk** [comment]
Oh, sorry. Yeah that's weird.

Ping @joecohens any insight with CORS?

**Comment by jbrooksuk** [comment]
@GrahamCampbell any ideas with this?

**Comment by GrahamCampbell** [comment]
This is perhapse caused by the missing headers on the other website, not on the target we're calling.

**Comment by GrahamCampbell** [comment]
Oh, I think see what's happening here. Can you confirm, does the browser first send a HEAD request, and then decide to do nothing?

**Comment by palktonderAtpowelno** [comment]
I still have this issue so this case should not be closed.

**Comment by palktonderAtpowelno** [comment]
I'm not sure if I should create a new issue for this problem since this issue is being automatically closed. For me this seems pretty simple to reproduce: Call https://demo.cachethq.io//api/v1/components from Postman (GET or OPTIONS). Observe that there is no Access-Control-Allow-Origin header, thereby effectively making it impossible to use from an external web site
